test/functional/rpc_fundrawtransaction.py

```diff
@@ -1200,6 +1201,67 @@ def test_add_inputs_default_value(self):
 
         self.nodes[2].unloadwallet("test_preset_inputs")
 
+    def test_preset_inputs_selection_crash(self):
+        self.log.info('Test wallet preset inputs selection')
+
+        # This exercises the bug presented in #26559 that
+        # ended up on (1) a wallet crash or (2) a created and
+        # broadcasted tx that pays an insane fee amount.
+        #
+        # This is covered by making the wallet selects the preset
+        # inputs twice during the coin selection process (which
+        # is a dangerous bug on v24.0).
+        #
+        # Making the wallet think that the selection process result covers
+        # the entire tx target when it does not. It's actually creating
+        # a tx that sends more coins than what inputs are covering for.
+        #
+        # Which, combined with the subtract fee from outputs option,
+        # makes the wallet generate and broadcast a tx that pays
+        # up to the sum of all the preset inputs, minus one, amounts in
+        # fee :(.
+
+        # Create and fund the wallet with 2 UTXO of 5 BTC each
+        self.nodes[2].createwallet("test_preset_inputs_selection")
+        wallet = self.nodes[2].get_wallet_rpc("test_preset_inputs_selection")
+        outputs = {}
+        for _ in range(2):
+            outputs[wallet.getnewaddress(address_type="bech32")] = 5
+        self.nodes[0].sendmany("", outputs)
+        self.generate(self.nodes[0], 1)
+
+        # Select the preset inputs
+        coins = wallet.listunspent()
+        preset_inputs = [coins[0], coins[1]]
+
+        # Now let's create the tx
+        options = {
+            "inputs": [
+                {
+                    "txid": preset_inputs[0]["txid"],
+                    "vout": preset_inputs[0]["vout"]
+                },
+                {
+                    "txid": preset_inputs[1]["txid"],
+                    "vout": preset_inputs[1]["vout"]
+                },
+            ],
+            "add_inputs": True,
+            "subtract_fee_from_outputs": [0],
+            "add_to_wallet": False
+        }
+
+        # First case, use 'subtract_fee_from_outputs' to make the wallet create a tx that pays an insane fee on v24.0.
+        # (further details in #26559).
+        assert_raises_rpc_error(-4, "Insufficient funds", wallet.send, outputs=[{wallet.getnewaddress(address_type="bech32"): 14}], options=options)
```

Comment:
Cherry-picking this test on top of 24.x, I can see that the wallet mistakenly uses a preset input twice and deducts the fee+5BTC from the payment, but the fee is actually exactly what is expected (since at the end the transaction only has 10BTC inputs and not 15): https://github.com/glozow/bitcoin/blob/test-n26560/test/functional/rpc_fundrawtransaction.py#L1250-L1266

@achow101 suggested that maybe `nFeeRet` from `FundTransaction()` or  `CreatedTransactionResult::fee` could be incorrect but I wasn't able to get a test to show that. @achow101 then pointed me to this code showing the sffo fee should be correctly set before `CreateTransaction` returns:
https://github.com/bitcoin/bitcoin/blob/f668a3a85933512e7efc369b5b2922d44b4bad82/src/wallet/spend.cpp#L971-L1004

I could be mistaken, but in that case could you maybe write a test for v24.0 that shows what the expected vs actual fee are? Otherwise I'd suggest removing this from the comments/commit message/PR descriptions.

Having 2 test cases for sffo and non-sffo codepaths is still meaningful, but I think the current comment is misleading and can cause us to have trouble understanding this test in the future.

Reply:
Done, updated per feedback. Thanks.

Some context:
The "insane fee" label came from the `nFeeRet` calculation. Which, in case of this bug, sets a high value than is deducted from the recipient's amount. Which.. as we spoke via dm, it's true that this is internal to the wallet only.. so the description should be something like "double-counted preset inputs during Coin Selection incorrectly deduces the user's selected recipient's amount" (or something similar)